clients/src/test/java/org/apache/kafka/clients/admin/MockAdminClient.java

```diff
@@ -719,6 +719,12 @@ synchronized public ListConsumerGroupsResult listConsumerGroups(ListConsumerGrou
         throw new UnsupportedOperationException("Not implemented yet");
     }
 
+    @Override
+    public ListShareGroupsResult listShareGroups(ListShareGroupsOptions options) {
+        // Implementation will be done as part of future PRs in KIP-932
+        return new ListShareGroupsResult();
```

Comment:
If I am not wrong them you would require MockAdminClient for unit tests and this can be implemented.

Reply:
Yes, it could copy `listClientMetricsResources` as a starting point.

Reply:
I would copy the implementation of `listConsumerGroups()` for now. That just throws an exception. When we need mocking of this method, we will have to do a bit more work.